Source/Pawnmorphs/Esoteria/UserInterface/Settings/Dialog_BlacklistAnimal.cs

```diff
@@ -68,23 +77,42 @@ public override void DoWindowContents(Rect inRect)
 		{
 			float curY = 0;
 			Text.Font = GameFont.Medium;
-			Widgets.Label(new Rect(0, curY, inRect.width, Text.LineHeight), HEADER_TEXT);
-
-			curY += Text.LineHeight;
+			Widgets.Label(0, ref curY, inRect.width, HEADER_TEXT);
 
 			Text.Font = GameFont.Small;
-			Rect descriptionRect = new Rect(0, curY, inRect.width, 60);
-			Widgets.Label(descriptionRect, DESCRIPTION_TEXT);
-
-			curY += descriptionRect.height;
-			curY += 35;
+			Widgets.Label(0, ref curY, inRect.width, DESCRIPTION_TEXT);
+			curY += SPACER_SIZE;
 
 			float totalHeight = inRect.height - curY - Math.Max(APPLY_BUTTON_SIZE.y, Math.Max(RESET_BUTTON_SIZE.y, CANCEL_BUTTON_SIZE.y)) - SPACER_SIZE;
 			_animalListBox.Draw(inRect, 0, curY, totalHeight, (item, listing) =>
 			{
-				bool current = _canBeFormerHumanDictonary.TryGetValue(item, false);
-				listing.CheckboxLabeled(item.LabelCap, ref current, item.modContentPack.ModMetaData.Name);
-				_canBeFormerHumanDictonary[item] = current;
+				FormerHumanRestrictions current = _canBeFormerHumanDictonary[item];
+				
+				string tooltip = "";
+				string source = item.modContentPack?.ModMetaData?.Name;
+				if (String.IsNullOrWhiteSpace(source) == false)
+				{
+					tooltip += "Added by " + source;
```

Comment:
Just a minor thing, but this should be a translatable string.  I don't know if you're already aware of this, but just in case you aren't, you can parameterize language strings like so:

```
<PM_AddedBy>Added by {0}</PM_AddedBy>
```

And fill in the parameters by calling .Translate() like so:

```
"PM_AddedBy".Translate(source);
```

Reply:
I have done it everywhere else. This very pull request has a lot of translation keys involved.

But I have indeed missed this one. Good catch!

Reply:
Yeah I saw, just wasn't sure if you knew that they could take parameters!  Took me some time to discover that trick myself, so I figured I'd mention it just in case.

Reply:
Ohh missed that that was what you were showing